src/Microsoft.NuGet.Build.Tasks/ResolveNuGetPackageAssets.cs

```diff
@@ -932,17 +932,21 @@ private void AddReferencedPackages(JArray packageDependencies, SortedSet<string>
             }
         }
 
-        private string GetNuGetPackagePath(string packageId, string packageVersion)
+        private string GetNuGetPackagePath(string packageId, string packageVersion, string packageRelativePath)
         {
+            string relativePathToUse = String.IsNullOrEmpty(packageRelativePath)
+                                            ? Path.Combine(packageId, packageVersion)
+                                            : packageRelativePath.Replace('/', Path.DirectorySeparatorChar);
+
             foreach (var packagesFolder in _packageFolders)
             {
-                string packagePath = Path.Combine(packagesFolder, packageId, packageVersion);
+                string packageFullPath = Path.Combine(packagesFolder, relativePathToUse);
 
                 // The proper way to check if a package is available is to look for the hash file, since that's the last
                 // file written as a part of the restore process. If it's not there, it means something failed part way through.
-                if (_fileExists(Path.Combine(packagePath, $"{packageId}.{packageVersion}.nupkg.sha512")))
+                if (_fileExists(Path.Combine(packageFullPath, $"{packageId}.{packageVersion}.nupkg.sha512")))
```

Comment:
Case matters for this name also, if the relativePath is all lower then this file name will be all lower also.

For reference:
https://github.com/NuGet/NuGet.Client/blob/74f7d7557c089431e10ce55d0dce4cab8c36a54f/src/NuGet.Core/NuGet.Packaging/VersionFolderPathResolver.cs#L118